cachi2/core/package_managers/pip.py

```diff
@@ -1765,47 +1795,45 @@ def _process_package_distributions(
             else:
                 log.info("Filtering out %s due to checksum mismatch", wheel.filename)
 
-        if len(filtered) > 0 and len(wheels) == 0:
+        if len(filtered) == 0 and len(sdists) == 0:
+            raise PackageRejected(
+                f"No archives found for package {name}=={version}",
+                solution=(
+                    "It seems that this version does not exist or isn't published.\n"
+                    "You may be able to specify the dependency directly via a URL instead, "
+                    "for example the wheel for a GitHub release."
+                ),
+                docs=PIP_NO_SDIST_DOC,
```

Comment:
Do GitHub releases of Python projects usually include wheel files? I don't think so

The "solution" here would probably be to double-check if the name and version that the user specified really exists on PyPI